diff -urN u-boot-2017.05/include/configs/usbarmory.h u-boot-2017.05-usbarmory_buildroot/include/configs/usbarmory.h
--- u-boot-2017.05/include/configs/usbarmory.h	2017-05-08 16:11:08.000000000 +0200
+++ u-boot-2017.05-usbarmory_buildroot/include/configs/usbarmory.h	2017-08-02 17:55:18.465526340 +0200
@@ -77,7 +77,7 @@
 
 #define MEM_LAYOUT_ENV_SETTINGS			\
 	"kernel_addr_r=0x70800000\0"		\
-	"fdt_addr_r=0x71000000\0"		\
+	"fdt_addr_r=0x72000000\0"		\
 	"scriptaddr=0x70800000\0"		\
 	"pxefile_addr_r=0x70800000\0"		\
 	"ramdisk_addr_r=0x73000000\0"
