From 4b4ef79020fce4122ff92305088716625d119815 Mon Sep 17 00:00:00 2001
From: Andrej Rosano <andrej@inversepath.com>
Date: Wed, 8 Jun 2016 14:53:18 +0200
Subject: [PATCH 1/2] ARM: mx53: add support for HAB commands

Signed-off-by: Andrej Rosano <andrej@inversepath.com>
---
 arch/arm/imx-common/hab.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/imx-common/hab.c b/arch/arm/imx-common/hab.c
index 8bbcc22..201b260 100644
--- a/arch/arm/imx-common/hab.c
+++ b/arch/arm/imx-common/hab.c
@@ -264,6 +264,9 @@ uint8_t hab_engines[16] = {
 
 bool is_hab_enabled(void)
 {
+#ifdef CONFIG_MX53
+	return true;
+#else
 	struct imx_sec_config_fuse_t *fuse =
 		(struct imx_sec_config_fuse_t *)&imx_sec_config_fuse;
 	uint32_t reg;
@@ -276,6 +279,7 @@ bool is_hab_enabled(void)
 	}
 
 	return (reg & IS_HAB_ENABLED_BIT) == IS_HAB_ENABLED_BIT;
+#endif
 }
 
 static inline uint8_t get_idx(uint8_t *list, uint8_t tgt)
@@ -364,6 +368,7 @@ int get_hab_status(void)
 	return 0;
 }
 
+#ifdef CONFIG_MX6
 uint32_t authenticate_image(uint32_t ddr_start, uint32_t image_size)
 {
 	uint32_t load_addr = 0;
@@ -382,8 +387,11 @@ uint32_t authenticate_image(uint32_t ddr_start, uint32_t image_size)
 	if (is_hab_enabled()) {
 		printf("\nAuthenticate image from DDR location 0x%x...\n",
 		       ddr_start);
+#endif
 
+#ifdef CONFIG_MX6
 		hab_caam_clock_enable(1);
+#endif
 
 		if (hab_rvt_entry() == HAB_SUCCESS) {
 			/* If not already aligned, Align to ALIGN_SIZE */
@@ -454,7 +462,9 @@ uint32_t authenticate_image(uint32_t ddr_start, uint32_t image_size)
 			puts("hab entry function fail\n");
 		}
 
+#ifdef CONFIG_MX6
 		hab_caam_clock_enable(0);
+#endif
 
 		get_hab_status();
 	} else {
@@ -479,6 +489,7 @@ int do_hab_status(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 	return 0;
 }
 
+#ifdef CONFIG_MX6
 static int do_authenticate_image(cmd_tbl_t *cmdtp, int flag, int argc,
 				char * const argv[])
 {
@@ -495,6 +506,7 @@ static int do_authenticate_image(cmd_tbl_t *cmdtp, int flag, int argc,
 
 	return rcode;
 }
+#endif
 
 U_BOOT_CMD(
 		hab_status, CONFIG_SYS_MAXARGS, 1, do_hab_status,
@@ -502,6 +514,7 @@ U_BOOT_CMD(
 		""
 	  );
 
+#ifdef CONFIG_MX6
 U_BOOT_CMD(
 		hab_auth_img, 3, 0, do_authenticate_image,
 		"authenticate image via HAB",
@@ -509,3 +522,4 @@ U_BOOT_CMD(
 		"addr - image hex address\n"
 		"ivt_offset - hex offset of IVT in the image"
 	  );
+#endif
-- 
2.8.3

